core/src/jsMain/kotlin/SharedRepeatableTask.kt

```diff
@@ -1,44 +1,35 @@
 package com.juul.kable
 
 import kotlinx.coroutines.CoroutineScope
-import kotlinx.coroutines.Job
+import kotlinx.coroutines.Deferred
+import kotlinx.coroutines.async
 import kotlinx.coroutines.cancelAndJoin
-import kotlinx.coroutines.launch
 
 /**
  * A mechanism for launching and awaiting some shared job repeatedly.
  *
- * The job is launched by calling [launchAsync]. Subsequent calls to [launchAsync] will return the
- * same (i.e. shared) job if it is still executing.
+ * The job is launched by calling [getOrLaunchAsync]. Subsequent calls to [getOrLaunchAsync] will return the
+ * same (i.e. shared) [job][Deferred] if it is still executing.
  */
-internal class SharedRepeatableTask(
+internal class SharedRepeatableTask<T>(
     private val scope: CoroutineScope,
-    private val task: suspend CoroutineScope.() -> Unit
+    private val task: suspend CoroutineScope.() -> T
 ) {
 
-    private var job: Job? = null
+    private var deferred: Deferred<T>? = null
+
+    /** If there is a running [job][Deferred] returns it. Otherwise, launches and returns a new [job][Deferred]. */
+    fun getOrLaunchAsync() = deferred
```

Comment:
I think the `Launch` in the name is misleading?
Perhaps just:

```suggestion
    fun getOrAsync() = deferred
```